src/render-instruction.ts

```diff
@@ -179,18 +185,27 @@ ${typeMapperImports.join('\n')}`.trim()
         : ''
 
     const requiredKeys = requireds
-      .map(({ name, isMut, isSigner, knownPubkey }) => {
-        const pubkey =
-          knownPubkey == null
-            ? `accounts.${name}`
-            : `accounts.${name} ?? ${renderKnownPubkeyAccess(
-                knownPubkey,
-                this.programIdPubkey
-              )}`
+      .map(({ name, isMut, isSigner, knownPubkey, optional }) => {
```

Comment:
This is starting to get hairy.
Maybe we should have two `renderIxAccountKey` methods even if that means duplicating a bit of code in order to not have these branches for the different optional account strategies all over.

Or maybe at the very least let's pull each one into it's own method/function so this gets easier to parse.